webextension/background/auth.js

```diff
@@ -78,55 +84,58 @@ this.auth = (function() {
   function login(options) {
     const { ownershipCheck, noRegister } = options || {};
     return new Promise((resolve, reject) => {
-      const loginUrl = main.getBackend() + "/api/login";
-      // TODO: replace xhr with Fetch #2261
-      const req = new XMLHttpRequest();
-      req.open("POST", loginUrl);
-      req.onload = catcher.watchFunction(() => {
-        if (req.status === 404) {
-          if (noRegister) {
-            resolve(false);
+      return fetchStoredInfo.then(() => {
+        const registrationInfo = getRegistrationInfo();
```

Comment:
☝️ change to generate device on login when not in local storage
👇 white space changes